pkg/receive/handler.go

```diff
@@ -976,6 +1011,16 @@ func determineWriteErrorCause(err error, threshold int) error {
 	return err
 }
 
+// errWithReplicaFunc is a type to enable sending replica number and err over channel.
+type errWithReplicaFunc func() (uint64, error)
```

Comment:
Why does this have to a function and can't be a `struct { uint64, error }`?

Reply:
Does not have to, I found it simpler to "send" it via function, no strong preference here though.

Reply:
If we don't need a function, I think we should "downgrade" the type to a struct to be as simple as possible.